tests/core/test_dataset.py

```diff
@@ -53,7 +54,9 @@ def test_invalid_copy_from(grid):
 def n_numbers(draw, n):
     numbers = []
     for _ in range(n):
-        number = draw(one_of(floats(), integers()))
+        number = draw(
+            one_of(floats(), integers(max_value=np.iinfo(int).max, min_value=np.iinfo(int).min))
+        )
```

Comment:
Using `integers()` will test integer over/underflow, e.g. values such as `np.iinfo(int).min-1`. This is great for test coverage, but it also revealed that `numpy` will actually create an `object` array instead of an `int` array in some cases. E.g.:
``` python
import numpy as np
np.array((np.iinfo(int).min-1))
# array(-9223372036854775809, dtype=object)
```

And so because it's an object array, the `_cast_to_numpy` function for the array validation raises a TypeError. This causes the tests to fail. And I don't know how to add an exception to this, e.g. how to add something like `pytest.xfail` using `hypothesis`. Instead, max and min and for integers is restricted so there is no over/underflow.